packages/language/src/document.js

```diff
@@ -634,11 +706,160 @@ Document.prototype.detectSentiment = function(options, callback) {
+ * document.detectSyntax(function(err, syntax) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // syntax = {
+ *   //   sentences: [
+ *   //     'Google is an American multinational technology company ' +
+ *   //     'specializing in Internet-related services and products.'
+ *   //   ],
+ *   //   tokens: [
+ *   //     {
+ *   //       text: 'Google',
+ *   //       partOfSpeech: 'Noun (common and proper)'
+ *   //     },
+ *   //     {
+ *   //       text: 'is',
+ *   //       partOfSpeech: 'Verb (all tenses and modes)'
+ *   //     },
+ *   //     ...
+ *   //   ],
+ *   //   language: 'en'
+ *   // }
+ * });
+ *
+ * //-
+ * // Verbose mode may also be enabled for more detailed results.
+ * //-
+ * var options = {
+ *   verbose: true
+ * };
+ *
+ * document.detectSyntax(options, function(err, syntax) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // syntax = {
+ *   //   sentences: [
+ *   //     {
+ *   //       text: {
+ *   //         content:
+ *   //           'Google is an American multinational technology company' +
+ *   //           'specializing in Internet-related services and products.',
+ *   //         beginOffset: -1
+ *   //       },
+ *   //       sentiment: {
+ *   //         score: 100
+ *   //         magnitude: 4
+ *   //       }
+ *   //     }
+ *   //   ],
+ *   //   tokens: [
+ *   //     {
+ *   //       text: {
+ *   //         content: 'Google',
+ *   //         beginOffset: -1
+ *   //       },
+ *   //       partOfSpeech: {
+ *   //         tag: 'NOUN',
+ *   //         aspect: 'PERFECTIVE',
+ *   //         case: 'ADVERBIAL',
+ *   //         form: 'ADNOMIAL',
+ *   //         gender: 'FEMININE',
+ *   //         mood: 'IMPERATIVE',
+ *   //         number: 'SINGULAR',
+ *   //         person: 'FIRST',
+ *   //         proper: 'PROPER',
+ *   //         reciprocity: 'RECIPROCAL',
+ *   //         tense: 'PAST',
+ *   //         voice: 'PASSIVE'
+ *   //       },
+ *   //       dependencyEdge: {
+ *   //         headTokenIndex: 1,
+ *   //         label: 'NSUBJ'
+ *   //       },
+ *   //       lemme: 'Google'
+ *   //     }
+ *   //   ],
+ *   //   language: 'en'
+ *   // }
+ * });
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * document.detectSyntax().then(function(data) {
+ *   var syntax = data[0];
+ *   var apiResponse = data[1];
+ * });
+ */
+Document.prototype.detectSyntax = function(options, callback) {
+  if (is.fn(options)) {
+    callback = options;
+    options = {};
+  }
+
+  var verbose = options.verbose === true;
+
+  this.api.Language.analyzeSyntax({
+    document: this.document,
+    encodingType: this.encodingType
+  }, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var originalResp = extend(true, {}, resp);
+
+    var syntax = {
+      sentences: Document.formatSentences_(resp.sentences, verbose),
+      tokens: Document.formatTokens_(resp.tokens, verbose),
```

Comment:
@stephenplusplus since `detectSentiment` only returns sentences & language via verbose mode, should we follow that convention and only return the tokens by default here?